src/Eloquent/Model.php

```diff
@@ -328,12 +328,22 @@ public function attributesToArray()
         // MongoDB related objects to a string representation. This kind
         // of mimics the SQL behaviour so that dates are formatted
         // nicely when your models are converted to JSON.
-        foreach ($attributes as $key => &$value) {
+        $convertMongoObjects = function (&$value) use (&$convertMongoObjects) {
```

Comment:
This should be a private method.

Reply:
Ok now is split

Reply:
For the test, I simply reused existing templates. Would you like me to create a new one?
Yes for nested fields (with no integration relationship), but will it work without my patch for more deep level fields?